standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java

```diff
@@ -374,9 +362,14 @@ public HMSHandler(String name, Configuration conf, boolean init) throws MetaExce
         }
       }
     }
-    if (init) {
-      init();
-    }
+  }
+
+  @VisibleForTesting
+  public static HMSHandler getInitializedHandler(String name, Configuration conf)
```

Comment:
I don't understand why would we need this method - we already had the `init` boolean trick

I think you are after making sure that all HMSHandler are initialized; without the patch didn't you get NPE-s when you try to access databaseCount - when the handler is not initialized?

Reply:
The `init` is set to true only in our tests,  the [RetryingHMSHandler](https://github.com/apache/hive/blob/master/standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/RetryingHMSHandler.java#L80) will init the HMSHandler,  and HiveMetaStore(including the local mode) uses RetryingHMSHandler to delegate the request to the initialized HMSHandler, so I propose to remove the boolean trick in the constructors, introduces a new method `getInitializedHandler` mainly for the tests.

Reply:
Hey @kgyrtkirk, any thoughts or comments?